buildSrc/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionArchiveCheckPlugin.java

```diff
@@ -0,0 +1,311 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.internal;
+
+import org.elasticsearch.gradle.LoggedExec;
+import org.elasticsearch.gradle.VersionProperties;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.BasePlugin;
+import org.gradle.api.specs.Spec;
+import org.gradle.api.tasks.TaskProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+
+public class InternalDistributionArchiveCheckPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.getPlugins().apply(BasePlugin.class);
+        String buildTaskName = calculateBuildTask(project.getName());
+        Task buildDistTask = project.getParent().getTasks().getByName(buildTaskName);
+        DistributionArchiveCheckExtension distributionArchiveCheckExtension = project.getExtensions()
+            .create("distributionArchiveCheck", DistributionArchiveCheckExtension.class);
+
+        File archiveExtractionDir = calculateArchiveExtractionDir(project);
+        Spec<Task> toolAvailableSpec = task -> toolExists(project);
+
+        // sanity checks if archives can be extracted
+        TaskProvider<LoggedExec> checkExtraction = registerCheckExtractionTask(
+            project,
+            buildDistTask,
+            archiveExtractionDir,
+            toolAvailableSpec
+        );
+        TaskProvider<Task> checkLicense = registerCheckLicenseTask(
+            project,
+            buildDistTask,
+            archiveExtractionDir,
+            toolAvailableSpec,
+            checkExtraction
+        );
+
+        TaskProvider<Task> checkNotice = registerCheckNoticeTask(
+            project,
+            buildDistTask,
+            archiveExtractionDir,
+            toolAvailableSpec,
+            checkExtraction
+        );
+        TaskProvider<Task> checkTask = project.getTasks().named("check");
+        checkTask.configure(task -> {
+            task.dependsOn(checkExtraction);
+            task.dependsOn(checkLicense);
+            task.dependsOn(checkNotice);
+        });
+
+        if (project.getName().contains("zip") || project.getName().contains("tar")) {
```

Comment:
This is actually a change in behaviour from master as discussed here: https://elastic.slack.com/archives/C8UUBNASY/p1598283630030100